packages/builders/src/messages/embed/Embed.ts

```diff
@@ -13,29 +13,29 @@ import {
 	urlPredicate,
 	validateFieldLength,
 } from './Assertions';
-import { AuthorOptions, FooterOptions, UnsafeEmbed } from './UnsafeEmbed';
+import { EmbedAuthorOptions, EmbedFooterOptions, UnsafeEmbed } from './UnsafeEmbed';
 
 /**
- * Represents an embed in a message (image/video preview, rich embed, etc.)
+ * Represents a validated embed in a message (image/video preview, rich embed, etc.)
  */
 export class Embed extends UnsafeEmbed {
 	public override addFields(...fields: APIEmbedField[]): this {
 		// Ensure adding these fields won't exceed the 25 field limit
-		validateFieldLength(this.fields, fields.length);
+		validateFieldLength(this.fields ?? [], fields.length);
```

Comment:
`validateFieldLength` should be tweaked to support undefined as first parameter, so no extra allocations are needed.